Code/Common/src/sitkCreateInterpolator.hxx

```diff
@@ -0,0 +1,269 @@
+    if (p == nullptr) {
+      return p;
+    }
+    p->SetSigma(sigma);
+    //p->SetAlpha(4.0);
+    return RType(p);
+  }
+  case sitkLabelGaussian: {
+    using InterpolatorType = itk::LabelImageGaussianInterpolateImageFunction<TImageType, double>;
+
+    typename InterpolatorType::ArrayType sigma;
+
+    for( unsigned int i = 0; i < TImageType::ImageDimension; ++i )
+      {
+      sigma[i] = spacing[i];
+      }
+    typename InterpolatorType::Pointer p = InterpolatorType::New();
+    p->SetSigma(sigma);
+    p->SetAlpha(1.0);
+    return RType(p);
+  }
+  case sitkHammingWindowedSinc: {
+
+    using WindowFunction =
+        typename itk::Function::HammingWindowFunction<WindowingRadius, double,
+                                                      double>;
+    using InterpolatorType = itk::WindowedSincInterpolateImageFunction<
+        TImageType, WindowingRadius, WindowFunction, BoundaryCondition>;
+    return RType(ConditionalCreateInterpolator<InterpolatorType>(
+        typename IsBasic<TImageType>::Type()));
+  }
+  case sitkCosineWindowedSinc: {
+    using WindowFunction =
+        typename itk::Function::CosineWindowFunction<WindowingRadius, double,
+                                                     double>;
+    using InterpolatorType = itk::WindowedSincInterpolateImageFunction<
+        TImageType, WindowingRadius, WindowFunction, BoundaryCondition>;
+    return RType(ConditionalCreateInterpolator<InterpolatorType>(
+        typename IsBasic<TImageType>::Type()));
+  }
+  case sitkWelchWindowedSinc: {
+    using WindowFunction =
+        typename itk::Function::WelchWindowFunction<WindowingRadius, double,
+                                                    double>;
+    using InterpolatorType = itk::WindowedSincInterpolateImageFunction<
+        TImageType, WindowingRadius, WindowFunction, BoundaryCondition>;
+    return RType(ConditionalCreateInterpolator<InterpolatorType>(
+        typename IsBasic<TImageType>::Type()));
+  }
+  case sitkLanczosWindowedSinc: {
+    using WindowFunction =
+        typename itk::Function::LanczosWindowFunction<WindowingRadius, double,
+                                                      double>;
+    using InterpolatorType = itk::WindowedSincInterpolateImageFunction<
+        TImageType, WindowingRadius, WindowFunction, BoundaryCondition>;
+    return RType(ConditionalCreateInterpolator<InterpolatorType>(
+        typename IsBasic<TImageType>::Type()));
+  }
+  case sitkBlackmanWindowedSinc: {
+    using WindowFunction =
+        typename itk::Function::BlackmanWindowFunction<WindowingRadius, double,
+                                                       double>;
+    using InterpolatorType = itk::WindowedSincInterpolateImageFunction<
+        TImageType, WindowingRadius, WindowFunction, BoundaryCondition>;
+    return RType(ConditionalCreateInterpolator<InterpolatorType>(
+        typename IsBasic<TImageType>::Type()));
+  }
+  case sitkBSplineResamplerOrder1: {
```

Comment:
Not sure if this should support resampling based on the itk::BSplineResampleImageFunction because it is specific to BSpline coefficient images and should not be used for regular images. On the other hand this is consistent with the ResampleImageFilter.